src/bin/ipa_bench/gen_events.rs

```diff
@@ -258,54 +176,44 @@ fn gen_events<R: RngCore + CryptoRng>(
 
     let mut last_conversion = last_impression;
 
-    for _ in 0..params.conversions {
+    for _ in 0..conversions {
         let conversion_value = sample.conversion_value_per_ad(rng);
         let t = last_conversion + sample.conversions_time_diff(rng);
 
-        if secret_share {
-            events.push(Event::EncryptedTrigger(ETriggerEvent {
-                event: EEvent {
-                    matchkeys: ss_mks.clone(),
-                    //TODO: Carry to next epoch if timestamp > DAYS_IN_EPOCH
-                    epoch: params.epoch,
-                    timestamp: u32::try_from(t.as_secs()).unwrap().xor_split(ss_rng),
-                },
-                value: conversion_value.xor_split(ss_rng),
-            }));
-        } else {
-            events.push(Event::Trigger(TriggerEvent::new(
-                matchkeys.clone(),
-                //TODO: Carry to next epoch if timestamp > DAYS_IN_EPOCH
-                params.epoch,
-                u32::try_from(t.as_secs()).unwrap(),
-                conversion_value,
-            )));
-        }
+        // days_in_epoch + sample.impression_time_diff < 4 years < u8::MAX (assuming 1 epoch = 1 week)
+        #[allow(clippy::cast_possible_truncation)]
+        let new_epoch = epoch + (t.as_secs() / SECONDS_IN_A_WEEK) as u8;
```

Comment:
you may want to encapsulate the epoch into a struct that handles addition/subtraction and backed by a u8

Reply:
i strongly agree. especially for cases like below, so you can hide such things as `u32::try_from(...).unwrap()`